configure.ac

```diff
@@ -303,8 +301,36 @@ AC_MSG_RESULT([${xslstylesheetsdir}])
 AC_MSG_CHECKING([git checkout])
 GIT_CHECKOUT="no"
 if test -n "${GIT}" -a -d "${srcdir}/.git"; then
-        AC_DEFINE([HAVE_CONFIG_VERSION_H], [1], [extra version available in config-version.h])
-        GIT_CHECKOUT="yes"
+	AC_DEFINE([HAVE_CONFIG_VERSION_H], [1], [extra version available in config-version.h])
+	GIT_CHECKOUT="yes"
+
+	REVISION_COMMIT="$(${GIT} log -1 --pretty=format:'%h')"
+
+	REVISION_DESCRIPTION="$(${GIT} describe || echo '<version not available>' )"
+	if test "${REVISION_DESCRIPTION}" = "<version not available>"; then
+		REVISION_DESCRIPTION="$(${GIT} describe --tags || echo '<version not available>')"
+	fi
+
+	if test "${REVISION_DESCRIPTION}" != "<version not available>"; then
+		NUMBER_OF_COMMITS="$(echo ${REVISION_DESCRIPTION} | sed -E 's/^.*-([[1-9]][[0-9]]+)-g[[A-Za-z0-9]]+/\1/')"
+
+		if test "${REVISION_DESCRIPTION}" = "${NUMBER_OF_COMMITS}"; then
+			NUMBER_OF_COMMITS="$(echo ${REVISION_DESCRIPTION} | sed -E 's/^.*-([[1-9]][[0-9]]+)/\1/')"
+		fi
+
+		if test "${REVISION_DESCRIPTION}" = "${NUMBER_OF_COMMITS}"; then
+			NUMBER_OF_COMMITS=0
+		fi
+	else
+		NUMBER_OF_COMMITS=0
+	fi
+
+	COMMIT_DATE="$(${GIT} log -1 --pretty=fuller --date=iso | grep CommitDate: | sed -E 's/^(CommitDate:\s.*)\s+.*/\1/')"
```

Comment:
Ok, thanks.

By default _'git describe'_ use the annotated tags and
for the current _master_ it composes name relative to _0.15.0_ tag.

For _master_ it's more appropriate then reference to _v0.16.0-pre1_ obtained with _'git rev-list --tags'_. This last tag had to be separated to 0.16.0 dedicated branch. It's not so important and will be fixed by the next release.